src/test/scala/com/exasol/cloudetl/kafka/KafkaConsumerFactoryTest.scala

```diff
@@ -1,24 +1,125 @@
-      "TABLE_NAME" -> "exasolTable"
+    val consumerProperties = KafkaConsumerProperties(defaultProperties)
+    val kafkaConsumer = KafkaConsumerFactory(consumerProperties, mock[ExaMetadata])
+    assert(kafkaConsumer.isInstanceOf[KafkaConsumer[String, GenericRecord]])
+  }
+
+  test("apply throws if required SCHEMA_REGISTRY_URL is not set") {
+    val properties = defaultProperties - ("SCHEMA_REGISTRY_URL")
+    val consumerProperties = KafkaConsumerProperties(properties)
+    val thrown = intercept[KafkaConnectorException] {
+      KafkaConsumerFactory(consumerProperties, mock[ExaMetadata])
+    }
+    assert(thrown.getMessage.contains("Error creating a Kafka consumer for topic 'topic'."))
+    assert(
+      thrown.getMessage.contains(
+        "Please provide a value for the SCHEMA_REGISTRY_URL property!"
+      )
     )
+  }
+
+  test("apply throws if required BOOTSTRAP_SERVERS is not set") {
+    val properties = defaultProperties - ("BOOTSTRAP_SERVERS")
     val consumerProperties = KafkaConsumerProperties(properties)
-    val kafkaConsumer = KafkaConsumerFactory(consumerProperties, mock[ExaMetadata])
+    val thrown = intercept[KafkaConnectorException] {
+      KafkaConsumerFactory(consumerProperties, mock[ExaMetadata])
+    }
+    assert(
+      thrown.getMessage.contains(
+        "Please provide a value for the BOOTSTRAP_SERVERS property!"
+      )
+    )
+  }
+
+  test("apply throws if secure SSL properties are provided without connection object") {
+    val properties = defaultProperties ++ Map(
+      "SSL_ENABLED" -> "true",
+      "SECURITY_PROTOCOL" -> "SSL",
+      "SSL_KEY_PASSWORD" -> "PASSWORD"
+    )
+    val consumerProperties = KafkaConsumerProperties(properties)
+    val thrown = intercept[KafkaConnectorException] {
+      KafkaConsumerFactory(consumerProperties, mock[ExaMetadata])
+    }
+    assert(
+      thrown.getMessage ===
+        "Please use a named connection object to provide secure SSL properties."
+    )
+  }
+
+  test("apply returns a SSL enabled kafka consumer") {
+    val kafkaConsumer = createSSLEnabledKafkaConsumer(DUMMY_KEYSTORE_FILE, DUMMY_TRUSTSTORE_FILE)
     assert(kafkaConsumer.isInstanceOf[KafkaConsumer[String, GenericRecord]])
   }
 
+  test("apply throws if SSL Keystore JKS file is not available") {
+    val thrown = intercept[KafkaConnectorException] {
+      createSSLEnabledKafkaConsumer(Paths.get("ssl_keystore_file"), DUMMY_TRUSTSTORE_FILE)
+    }
+    val msg = thrown.getMessage()
+    assert(msg.contains("Unable to find the SSL keystore file"))
+    assert(msg.contains("Please make sure it is successfully uploaded to BucketFS bucket"))
+  }
+
+  test("apply throws if SSL Truststore JKS file is not available") {
+    val thrown = intercept[KafkaConnectorException] {
+      createSSLEnabledKafkaConsumer(DUMMY_KEYSTORE_FILE, Paths.get("ssl_truststore_file"))
+    }
+    val msg = thrown.getMessage()
+    assert(msg.contains("Unable to find the SSL truststore file"))
```

Comment:
If you add error code to your error messages, you won't need to check the whole message, only the code will be enough

Reply:
Thanks @AnastasiiaSergienko, I have added #12 not to forget.